### PR TITLE
[PAR-WEBFT] MEYER / MORSLI - All valid

### PR DESCRIPTION
**Pull request by ameyerfr**

### CODE REVIEW

File: starter_code/src/movies.js [hunk 1]
```diff
@@ -1,15 +1,115 @@
...
+  // Sort movies of the same year by title
+  moviesYears.sort( (a,b) => {
+    if (a.year === b.year) {
+      return a.title === b.title ? 0 : a.title > b.title ? 1 : -1;
+    }
+  });
+
+  return moviesYears;
+}
+
```
Comment (ta-web-paris) [inline starter_code/src/movies.js]: Here you could chain everything at once, and especially, you can add more logic to your sort function.

moviesYears.sort((a,b) => {
if(a.year === b.year) return a..title.localeCompare(b.title)
return a.year - b.year;
}
Information on localeCompare: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare

File: starter_code/src/movies.js [hunk 2]
```diff
@@ -1,15 +1,115 @@
...
+  });
+
+  // Order the list alphabetically
+  moviesList.sort( (a,b) => {
+    return a === b ? 0 : a > b ? 1 : -1;
+  });
+
+  // Return first 20 elements
+  return moviesList.slice(0,20);
+}
```
Comment (ta-web-paris) [inline starter_code/src/movies.js]: Here again you could chain everything at once

```
function orderAlphabetically(movies) {
  return movies.map(m => m.title).sort((a,b) => a.localeCompare(b)).slice(0,20)
}
```

The map creates a new array, then your sort it, then you slice it (which returns a new array)

**Comment by ta-web-paris** [comment]
Well done ! closing the PR.

File: starter_code/src/movies.js (hunk 1 above)
Comment (ameyerfr) [reply starter_code/src/movies.js]: Agreed, this one was very verbose for no reason !

File: starter_code/src/movies.js (hunk 2 above)
Comment (ameyerfr) [reply starter_code/src/movies.js]: Agreed ! This one was very verbose for no reason.

**Pull request closed without merging**